src/clients/go/grpc_simple_client.go

```diff
@@ -0,0 +1,258 @@
+	inputData1 := inputs[1]
+
+	var inputBytes0 []byte
+	var inputBytes1 []byte
+	// Temp variable to hold our converted int32 -> []byte
+	bs := make([]byte, 4)
+	for i := 0; i < inputSize; i++ {
+		binary.LittleEndian.PutUint32(bs, inputData0[i])
+		inputBytes0 = append(inputBytes0, bs...)
+		binary.LittleEndian.PutUint32(bs, inputData1[i])
+		inputBytes1 = append(inputBytes1, bs...)
+	}
+
+	return [][]byte{inputBytes0, inputBytes1}
+}
+
+// Convert slice of 4 bytes to int32 (assumes Little Endian)
+func readInt32(fourBytes []byte) int32 {
+	buf := bytes.NewBuffer(fourBytes)
+	var retval int32
+	binary.Read(buf, binary.LittleEndian, &retval)
+	return retval
+}
+
+// Convert output's raw bytes into int32 data (assumes Little Endian)
+func Postprocess(inferResponse *triton.ModelInferResponse) [][]int32 {
+	var outputs []*triton.ModelInferResponse_InferOutputTensor
+	outputs = inferResponse.Outputs
+	output0 := outputs[0]
+	output1 := outputs[1]
+	outputBytes0 := output0.Contents.RawContents
+	outputBytes1 := output1.Contents.RawContents
+
+	outputData0 := make([]int32, outputSize)
+	outputData1 := make([]int32, outputSize)
+	for i := 0; i < outputSize; i++ {
+		outputData0[i] = readInt32(outputBytes0[i*4 : i*4+4])
+		outputData1[i] = readInt32(outputBytes1[i*4 : i*4+4])
+	}
+	return [][]int32{outputData0, outputData1}
+}
+
+func main() {
+	FLAGS := parseFlags()
+	fmt.Println("FLAGS:", FLAGS)
+
+	// Connect to gRPC server
+	conn, err := grpc.Dial(FLAGS.URL, grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("Couldn't connect to endpoint %s: %v", FLAGS.URL, err)
+	}
+	defer conn.Close()
+
+	// Create client from gRPC server connection
+	client := triton.NewGRPCInferenceServiceClient(conn)
+
+	serverLiveResponse := ServerLiveRequest(client)
+	fmt.Printf("Triton Health - Live: %v\n", serverLiveResponse.Live)
+
+	serverReadyResponse := ServerReadyRequest(client)
+	fmt.Printf("Triton Health - Ready: %v\n", serverReadyResponse.Ready)
+
+	modelMetadataResponse := ModelMetadataRequest(client, FLAGS.ModelName, "")
+	fmt.Println(modelMetadataResponse)
+
+	inputData0 := make([]uint32, inputSize)
+	inputData1 := make([]uint32, inputSize)
+	for i := 0; i < inputSize; i++ {
+		inputData0[i] = uint32(i)
+		inputData1[i] = 1
+	}
+	inputs := [][]uint32{inputData0, inputData1}
+	rawInput := Preprocess(inputs)
+
+	/* We use a simple model that takes 2 input tensors of 16 integers
+	each and returns 2 output tensors of 16 integers each. One
+	output tensor is the element-wise sum of the inputs and one
+	output is the element-wise difference. */
+	inferResponse := ModelInferRequest(client, rawInput, FLAGS.ModelName, FLAGS.ModelVersion)
+
+	/* We expect there to be 2 results (each with batch-size 1). Walk
+	over all 16 result elements and print the sum and difference
+	calculated by the model. */
+	outputs := Postprocess(inferResponse)
+	outputData0 := outputs[0]
+	outputData1 := outputs[1]
+
+	fmt.Println("\nChecking Inference Outputs\n--------------------------")
```

Comment:
Instead of just checking for this statement on our CI, can you validate the output data as well?

Reply:
The output data is being validated as a part of the test script

Reply:
I see. The PostProcess is comparing the values.